# Analysis of Variants

This folder contains the code related to the post-processing of the somatic (`somatic`) and germline (`germline`) variant files obtained with nf-core Sarek and nf-core OncoAnalyzer for normal versus tumor samples of the patient. These directories include the following files:

- `somatic`: This folder includes the code in Jupyter notebooks to perform the post-processing of the somatic variants.
  - `sarek_variants`: Reading and filtering variant data from original VCF files obtained with nf-core Sarek. Classify variant types and calculation of variant allele frequencies (VAF).
  - `oncoanalyzer_variants`: Reading and filtering variant data from original VCF files obtained with nf-core OncoAnalyzer. Classify variant types and calculation of variant allele frequencies (VAF).
  - `consensus_mut`: 
  - `clonality`:
  - `phylogeny`:


   Also includes the `mutation_rate.ipynb` that was used to calculate the 

- `germline`: This folder includes the code in Jupyter notebooks to perform the post-processing of the germline variants obtained with GATK Haplotypecaller tool from nf-core Sarek pipeline.



        > fdsafjal.ipynb <br>
        > fjajlfsf.ipynb <br>
      
    fasdfasfadd from [fafassd](https://www.ncbi.nlm.nih.gov/datasets/genome/GCF_000001405.26/). This Refsafjkaklja
