{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "c73fd91b-7519-4221-8833-b17424cf682d",
   "metadata": {},
   "source": [
    "# VEP conversion of common mutations table: germline sample"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "60362b67-8330-4d49-b4af-ad24bbcec963",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Needed basic packages\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "3b006fb6-2558-46eb-a09d-c382164ca413",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Import the list of germline mutations filtered by sequencing quality.\n",
    "\n",
    "mut_df = pd.read_csv('/workspace/projects/sjd_melos/MAFs_tables/Germline_VAF.tsv.gz', sep=\"\\t\", header=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "8e153107-7e98-4714-8473-cd13a38416b9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>patient1_AB9766</th>\n",
       "      <th>GT</th>\n",
       "      <th>AD</th>\n",
       "      <th>DP</th>\n",
       "      <th>GQ</th>\n",
       "      <th>PL</th>\n",
       "      <th>AD_ref</th>\n",
       "      <th>AD_alt</th>\n",
       "      <th>VAF</th>\n",
       "      <th>mut_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>13273</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>1444.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=1;AF=0.500;AN=2;BaseQRankSum=0.096;CNN_1D=-...</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>0/1:120,63:183:99:1452,0,3326</td>\n",
       "      <td>0/1</td>\n",
       "      <td>120,63</td>\n",
       "      <td>183.0</td>\n",
       "      <td>99</td>\n",
       "      <td>1452,0,3326</td>\n",
       "      <td>120</td>\n",
       "      <td>63.0</td>\n",
       "      <td>0.344262</td>\n",
       "      <td>SNV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>13813</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>114.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=1;AF=0.500;AN=2;BaseQRankSum=-5.261;CNN_1D=...</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>0/1:90,16:106:99:122,0,3424</td>\n",
       "      <td>0/1</td>\n",
       "      <td>90,16</td>\n",
       "      <td>106.0</td>\n",
       "      <td>99</td>\n",
       "      <td>122,0,3424</td>\n",
       "      <td>90</td>\n",
       "      <td>16.0</td>\n",
       "      <td>0.150943</td>\n",
       "      <td>SNV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>13838</td>\n",
       "      <td>rs200683566</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>503.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=1;AF=0.500;AN=2;BaseQRankSum=5.055;CNN_1D=-...</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>0/1:110,23:133:99:511,0,2959</td>\n",
       "      <td>0/1</td>\n",
       "      <td>110,23</td>\n",
       "      <td>133.0</td>\n",
       "      <td>99</td>\n",
       "      <td>511,0,2959</td>\n",
       "      <td>110</td>\n",
       "      <td>23.0</td>\n",
       "      <td>0.172932</td>\n",
       "      <td>SNV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14599</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>146.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=1;AF=0.500;AN=2;BaseQRankSum=-3.384;CNN_1D=...</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>0/1:158,15:173:99:154,0,6545</td>\n",
       "      <td>0/1</td>\n",
       "      <td>158,15</td>\n",
       "      <td>173.0</td>\n",
       "      <td>99</td>\n",
       "      <td>154,0,6545</td>\n",
       "      <td>158</td>\n",
       "      <td>15.0</td>\n",
       "      <td>0.086705</td>\n",
       "      <td>SNV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14604</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>167.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=1;AF=0.500;AN=2;BaseQRankSum=-1.322;CNN_1D=...</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>0/1:165,16:181:99:175,0,6793</td>\n",
       "      <td>0/1</td>\n",
       "      <td>165,16</td>\n",
       "      <td>181.0</td>\n",
       "      <td>99</td>\n",
       "      <td>175,0,6793</td>\n",
       "      <td>165</td>\n",
       "      <td>16.0</td>\n",
       "      <td>0.088398</td>\n",
       "      <td>SNV</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM    POS           ID REF ALT     QUAL FILTER  \\\n",
       "0  chr1  13273            .   G   C  1444.64   PASS   \n",
       "1  chr1  13813            .   T   G   114.64   PASS   \n",
       "2  chr1  13838  rs200683566   C   T   503.64   PASS   \n",
       "3  chr1  14599            .   T   A   146.64   PASS   \n",
       "4  chr1  14604            .   A   G   167.64   PASS   \n",
       "\n",
       "                                                INFO          FORMAT  \\\n",
       "0  AC=1;AF=0.500;AN=2;BaseQRankSum=0.096;CNN_1D=-...  GT:AD:DP:GQ:PL   \n",
       "1  AC=1;AF=0.500;AN=2;BaseQRankSum=-5.261;CNN_1D=...  GT:AD:DP:GQ:PL   \n",
       "2  AC=1;AF=0.500;AN=2;BaseQRankSum=5.055;CNN_1D=-...  GT:AD:DP:GQ:PL   \n",
       "3  AC=1;AF=0.500;AN=2;BaseQRankSum=-3.384;CNN_1D=...  GT:AD:DP:GQ:PL   \n",
       "4  AC=1;AF=0.500;AN=2;BaseQRankSum=-1.322;CNN_1D=...  GT:AD:DP:GQ:PL   \n",
       "\n",
       "                 patient1_AB9766   GT      AD     DP  GQ           PL  AD_ref  \\\n",
       "0  0/1:120,63:183:99:1452,0,3326  0/1  120,63  183.0  99  1452,0,3326     120   \n",
       "1    0/1:90,16:106:99:122,0,3424  0/1   90,16  106.0  99   122,0,3424      90   \n",
       "2   0/1:110,23:133:99:511,0,2959  0/1  110,23  133.0  99   511,0,2959     110   \n",
       "3   0/1:158,15:173:99:154,0,6545  0/1  158,15  173.0  99   154,0,6545     158   \n",
       "4   0/1:165,16:181:99:175,0,6793  0/1  165,16  181.0  99   175,0,6793     165   \n",
       "\n",
       "   AD_alt       VAF mut_type  \n",
       "0    63.0  0.344262      SNV  \n",
       "1    16.0  0.150943      SNV  \n",
       "2    23.0  0.172932      SNV  \n",
       "3    15.0  0.086705      SNV  \n",
       "4    16.0  0.088398      SNV  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mut_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "eeb10351-1ed1-4772-9457-265dc44b1d3a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# CHROM column has to maintain the # so VEP recognizes that this is a VCF file!!\n",
    "mut_df = mut_df.rename(columns = {'CHROM':'#CHROM'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "e8a7cded-e788-420d-9085-0c75093f113a",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['chr1',\n",
       " 'chr2',\n",
       " 'chr3',\n",
       " 'chr4',\n",
       " 'chr5',\n",
       " 'chr6',\n",
       " 'chr7',\n",
       " 'chr8',\n",
       " 'chr9',\n",
       " 'chr10',\n",
       " 'chr11',\n",
       " 'chr12',\n",
       " 'chr13',\n",
       " 'chr14',\n",
       " 'chr15',\n",
       " 'chr16',\n",
       " 'chr17',\n",
       " 'chr18',\n",
       " 'chr19',\n",
       " 'chr20',\n",
       " 'chr21',\n",
       " 'chr22',\n",
       " 'chrX',\n",
       " 'chrY']"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Perform a chromosome list\n",
    "chroms = []\n",
    "for n in range(1,23):\n",
    "    chroms.append('chr'+str(n))\n",
    "chroms.append('chrX')    \n",
    "chroms.append('chrY')  \n",
    "chroms\n",
    "\n",
    "# This is the same as previous code but in few lines\n",
    "# chroms = ['chr'+str(n) for n in range(1,23)  ]\n",
    "# chroms.append('chrX')\n",
    "# chroms.append('chrY')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "21f8adfd-5822-462d-bcae-9c554575535d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Extract the chromosomes data:\n",
    "\n",
    "path = '/workspace/projects/sjd_melos/vep/vep_input_files/germline/'\n",
    "\n",
    "for chrom in chroms:\n",
    "    file_name= path + chrom + '.tsv.gz'    \n",
    "    chrom_df = mut_df[mut_df['#CHROM']==chrom]\n",
    "    chrom_df.to_csv(file_name,sep='\\t', index=None, compression = \"gzip\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "melos",
   "language": "python",
   "name": "melos"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
